esp/platform/espp.cpp

```diff
@@ -432,6 +433,44 @@ static IPropertyTree * extractLegacyOptions(IPropertyTree * legacyOptions)
     return extractedOptions.getClear();
 }
 
+bool startEspEventRecording(const char * options, const char * filename)
+{
+    if (isEmptyString(options))
+        options = "threadid";
+
+    StringBuffer outputFilename;
+    const char * path = filename;
+    if (!isAbsolutePath(filename))
+    {
+        getTempFilePath(outputFilename, "eventrecorder", nullptr);
+        outputFilename.append(PATHSEPCHAR);
+        if (!isEmptyString(filename))
+        {
+            outputFilename.append(filename);
+        }
+        else
+        {
+            //MORE: Revisit this at a later date
+            unsigned seq = (unsigned)(get_cycles_now() % 100000);
+            outputFilename.append("espevents.").append((unsigned)GetCurrentProcessId()).append(".").append(seq).append(".evt");
+        }
+
+        path = outputFilename.str();
+        //MORE: The caller will need to know the full pathname
+    }
+
+    recursiveCreateDirectoryForFile(path);
+    if (!queryRecorder().startRecording(options, path, false))
+        return false;
+
+    return true;
+}
+
+bool stopEspEventRecording(EventRecordingSummary * optSummary)
```

Comment:
Will we use optSummary, or could the parameter be omitted? Roxie uses it with the control message response, but I'm not sure that applies here. If we wanted to log the summary, that could be handled in this method.

Reply:
I think when other code calls this function the summary will be used - e.g. an esp service to turn recording on and off.